retrieve_files.sh

```diff
@@ -104,51 +122,52 @@ case "$1" in
   log "ERR" "argument passed in is not valid: $1"
   exit 1
 esac
-     
-#create directory for downloaded files(if does not exist already)
-mkdir -p $dropLocation
 
-currentdate=$(date +%Y%m%d) 
+#create directory for downloaded files and logs(if does not exist already)
+mkdir -p $dropLocation
+mkdir -p $statusDirectory
 
 log "INFO" "-[JOB START] $(date): ------------"
 log "INFO" "================= Script starting ================="
 
-
+#if --all flag is set then get all available files from the page
 if $retrieveAll && ! $retrieveNone 
 then
   log "INFO" "Starting file download process"
   wget --directory-prefix=files/`date +%Y%m%d` -e robots=off --cut-dirs=3  --reject="index.html*" --no-parent --recursive --relative --level=1 --no-directories $baseURL
   log "INFO" "File download process complete"
+
+#use the start and end dates to determine which files to get
 elif ! $retrieveAll && ! $retrieveNone
 then
-  while [ $fileDate -lt $currentdate ]
+  log "INFO" "Starting file download process"
+  startDate=$(date '+%C%y%m%d' -d "$startDate -$(date -d $startDate +%u) days + 5 day")
+  while [ $startDate -lt $endDate ]
```

Comment:
Let me see if I follow the way this works... You increment the start date up by weeks, finding files named for the Friday of that week, until the start date is no longer less than the end date. Is that right?